Engine/Graphics/Direct3D/Render.cpp

```diff
@@ -3622,23 +3432,18 @@ void Render::DrawBuildingsD3D() {
             poly->uNumVertices = face.uNumVertices;
             poly->field_59 = 5;
 
-            v51 =
-                fixpoint_mul(-pOutdoor->vSunlight.x, face.pFacePlane.vNormal.x);
-            v53 =
-                fixpoint_mul(-pOutdoor->vSunlight.y, face.pFacePlane.vNormal.y);
-            v52 =
-                fixpoint_mul(-pOutdoor->vSunlight.z, face.pFacePlane.vNormal.z);
-            poly->dimming_level = 20 - fixpoint_mul(20, v51 + v53 + v52);
+            float f = (face.pFacePlaneOLD.vNormal.x / 65536.0) * pOutdoor->vSunlight.x + (face.pFacePlaneOLD.vNormal.y / 65536.0) * pOutdoor->vSunlight.y + (face.pFacePlaneOLD.vNormal.z / 65536.0) * pOutdoor->vSunlight.z;
+            poly->dimming_level = 20 - floorf(20.0 * f + 0.5f);
```

Comment:
just use std::round instead of floorf? Can drop this +0.5f then.

Reply:
Yes. Will also use float face plane instead of fixpoint plane.